Mathematics/Transformations/transformations.ipynb

```diff
@@ -0,0 +1,337 @@
+    "<img src=\"images/rotationimage.png\" width=\"400\" height=\"400\">\n",
+    "\n",
+    "### Non-Origin Point of Rotation\n",
+    "\n",
+    "When we have a point of rotation that is not at the origin we need to adjust the coordinate points. \n",
+    "\n",
+    "First translate the point of rotation to the origin, then do the same translation to all other points being rotated.\n",
+    "\n",
+    "For example if our point of rotation was at (1, 0) we would need to translate all the points horizontally by -1 for it to be at (0, 0). \n",
+    "\n",
+    "Once all the points have been rotated we would then do the opposite, translating all the points horizontally +1.\n",
+    "\n",
+    "For example, rotating a triangle around one of its points at the position (1, 1). \n",
+    "\n",
+    "> We start off with the triangle in its original position.\n",
+    ">\n",
+    "> <img src=\"images/triangle-rotation1.png\" width=\"500\" height=\"500\">\n",
+    ">\n",
+    "> Since we are rotating around the point (1, 1) we will translate the triangle by -1 on the *x* axis and -1 on the *y* axis.\n",
+    ">\n",
+    "> <img src=\"images/triangle-rotation2.png\" width=\"500\" height=\"500\">\n",
+    ">\n",
+    "> Now we can rotate the points by 90° to get the new coordinate points.\n",
+    ">\n",
+    "> <img src=\"images/triangle-rotation3.png\" width=\"500\" height=\"500\">\n",
+    ">\n",
+    "> The lines connecting the points are different colors so you can see each rotation is 90°. If we rotate the new triangle position by another 90° we will get the coordinates as if it had been rotated by 180° from the original position.\n",
+    ">\n",
+    "> <img src=\"images/triangle-rotation4.png\" width=\"500\" height=\"500\">\n",
+    ">\n",
+    "> If we rotate by 90° we will get the coordinates of the original triangle being rotated by 270°. Another 90° rotation would bring the orignal rotation to 360°, which is the same as not rotatingat all.\n",
+    ">\n",
+    "> <img src=\"images/triangle-rotation5.png\" width=\"500\" height=\"500\">\n",
+    ">\n",
+    "> The only thing we have left to do is to translate the triangle points by +1 on the *x* and +1 on the *y*, back to the original position.\n",
+    ">\n",
+    "> <img src=\"images/triangle-rotation6.png\" width=\"500\" height=\"500\">\n",
+    ">\n",
+    "> We now have the positions of the various rotations for the triangle in their actual positions.\n",
+    "\n",
+    "### Interactive Example\n",
+    "\n",
+    "In the interactive example below you can choose the adjust the degrees of rotation from 0° to 360° clockwise using the slider.\n",
+    "\n",
+    "The `rotate_points()` function could also be used in a different visualization for counterclockwise rotations or rotations greater than 360°."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def rotate_points(x, y, point, quarter_turns, direction):\n",
```

Comment:
Excellent feedback, thank you, I love your rewrite of the functions. Feel free to add the `translate_points` function to the first code cell if you'd like, it would be cleaner than my rewrite with list comprehension.